src/xb-exports.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+/* clang-cl also defines _MSC_VER */
+#if defined (_MSC_VER) && defined (LIBXMLB_BUILD_DLL)
+# define LIBXMLB_API __declspec (dllexport)
```

Comment:
this isn't ideal -- is there any way to change the behaviour so that anything in the .map file is exported? Otherwise we have to "export" the function in two places.

Reply:
Hi,

Yeah, I understand what you mean.  Perhaps I will try to see whether there is a way for a script to pull out the `xb_*` symbols from the .map file to generate a .def file (needed for Visual Studio-like builds, if not going by the `__declspec(dllexport)` route, or even update the script that is used to generate the .map file to be able to generate the .def file that is needed.

Reply:
Hi,

I think this is now ready as I added a script to go through the `libxmlb.map` file to generate the .def file that is now needed and used to export the symbols from the libxmlb DLL as we are now not going through the `__decspec(dllexport)` route.

With blessings, thank you!

Reply:
I've merged 3/5 of those patches, thanks. For the .def file -- would it be better to generate it from the introspection data rather than the .map file? It would be a lot less complex code (which I'm happy to write!) but it would mean depending on introspection for the clan-cl build.

Reply:
Hi,

The dilemma is, we would need the symbols to be exported from the DLL before we could build the introspection files, so this would become a circular situation…

Reply:
Right, understood.

Reply:
I ran the python file through black and merged it -- thanks.